mmdet/models/detectors/detection_transformer.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Dict, List, Tuple, Union
+
+from torch import Tensor
+
+from mmdet.registry import MODELS
+from mmdet.structures import OptSampleList, SampleList
+from mmdet.utils import ConfigType, OptConfigType, OptMultiConfig
+from .base import BaseDetector
+
+
+@MODELS.register_module()
+class TransformerDetector(BaseDetector):
+    """Base class for Transformer-based detectors.
+
+    Transformer-based detectors use an encoder to process output features of
```

Comment:
Transformer-based detectors use an encoder to process output features of backbone/neck,  then several queries interactive with encoded features and do the regression and classification.